{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "input_folder = \"input/\"\n",
    "output_folder = \"output/\"\n",
    "csv_separator = \",\"\n",
    "\n",
    "all_filenames = glob.glob(input_folder + \"*.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nome do arquivo:input/71890.csv\n",
      "nome do arquivo:input/71262.csv\n",
      "nome do arquivo:input/71974.csv\n",
      "nome do arquivo:input/71331.csv\n",
      "nome do arquivo:input/72025.csv\n",
      "nome do arquivo:input/71718.csv\n",
      "nome do arquivo:input/71213.csv\n",
      "nome do arquivo:input/71189.csv\n",
      "nome do arquivo:input/71950.csv\n",
      "nome do arquivo:input/71029.csv\n"
     ]
    }
   ],
   "source": [
    "for filename in all_filenames:\n",
    "    print(\"nome do arquivo:\" + filename)\n",
    "\n",
    "    sample_csv = pd.read_csv(filename , sep = csv_separator)\n",
    "    \n",
    "    #create column Variant Frequency 1 with the value from first position of patter xyz% -> abc%  \n",
    "    sample_csv['Variant Frequency 1'] = sample_csv['Variant Frequency'].str.split(' -> ').str[0]\n",
    "    #removes the % and convert to numeric\n",
    "    sample_csv['Variant Frequency 1'] = pd.to_numeric(sample_csv['Variant Frequency 1'].str.split('%').str[0])\n",
    "\n",
    "    #to the same as above, but for the second column\n",
    "    sample_csv['Variant Frequency 2'] = sample_csv['Variant Frequency'].str.split(' -> ').str[1]\n",
    "    sample_csv['Variant Frequency 2'] = sample_csv['Variant Frequency 2'].fillna('0')\n",
    "    sample_csv['Variant Frequency 2'] = pd.to_numeric(sample_csv['Variant Frequency 2'].str.split('%').str[0])\n",
    "\n",
    "    # filter data\n",
    "    sample_csv = sample_csv[sample_csv['gene'].notna()]\n",
    "    sample_csv = sample_csv[sample_csv['Protein Effect'] == \"Substitution\"]\n",
    "    sample_csv = sample_csv[sample_csv['Variant Frequency 1'] < 80.0 ] \n",
    "    \n",
    "    # calculate aa position (CDS Position / 3, rounded up)\n",
    "    sample_csv['AA Position'] = sample_csv['CDS Position'] / 3\n",
    "    sample_csv['AA Position'] = sample_csv['AA Position'].apply(np.ceil).astype(int)\n",
    "    \n",
    "    sample_csv = sample_csv.drop(columns=['Name', 'CDS', 'Change','Codon Change', 'Coverage', 'Polymorphism Type'])\n",
    "\n",
    "    # create output file\n",
    "    output_filename = output_folder + filename.split(\"/\")[1].split(\".\")[0] + \"_processada.csv\"\n",
    "    sample_csv.to_csv(output_filename, sep=';', encoding='utf-8')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>Amino Acid Change</th>\n",
       "      <th>CDS Position</th>\n",
       "      <th>Protein Effect</th>\n",
       "      <th>Variant Frequency</th>\n",
       "      <th>Unnamed: 11</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>Variant Frequency 1</th>\n",
       "      <th>Variant Frequency 2</th>\n",
       "      <th>AA Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>F -&gt; Y</td>\n",
       "      <td>15257</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>25.0%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>L -&gt; K</td>\n",
       "      <td>9595</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>28.2%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>L -&gt; K</td>\n",
       "      <td>9601</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>26.4%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>P -&gt; K</td>\n",
       "      <td>9598</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>26.4% -&gt; 26.9%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.4</td>\n",
       "      <td>26.9</td>\n",
       "      <td>3200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>ORF1ab</td>\n",
       "      <td>T -&gt; H</td>\n",
       "      <td>9604</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>26.2% -&gt; 26.4%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.2</td>\n",
       "      <td>26.4</td>\n",
       "      <td>3202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>M</td>\n",
       "      <td>Q -&gt; E</td>\n",
       "      <td>55</td>\n",
       "      <td>Substitution</td>\n",
       "      <td>75.7%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene Amino Acid Change  CDS Position Protein Effect Variant Frequency  \\\n",
       "43  ORF1ab            F -> Y         15257   Substitution             25.0%   \n",
       "46  ORF1ab            L -> K          9595   Substitution             28.2%   \n",
       "47  ORF1ab            L -> K          9601   Substitution             26.4%   \n",
       "48  ORF1ab            P -> K          9598   Substitution    26.4% -> 26.9%   \n",
       "50  ORF1ab            T -> H          9604   Substitution    26.2% -> 26.4%   \n",
       "63       M            Q -> E            55   Substitution             75.7%   \n",
       "\n",
       "   Unnamed: 11 Unnamed: 12 Unnamed: 13  Variant Frequency 1  \\\n",
       "43         NaN         NaN         NaN                 25.0   \n",
       "46         NaN         NaN         NaN                 28.2   \n",
       "47         NaN         NaN         NaN                 26.4   \n",
       "48         NaN         NaN         NaN                 26.4   \n",
       "50         NaN         NaN         NaN                 26.2   \n",
       "63         NaN         NaN         NaN                 75.7   \n",
       "\n",
       "    Variant Frequency 2  AA Position  \n",
       "43                  0.0         5086  \n",
       "46                  0.0         3199  \n",
       "47                  0.0         3201  \n",
       "48                 26.9         3200  \n",
       "50                 26.4         3202  \n",
       "63                  0.0           19  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_csv.tail(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
